addons/account_peppol/tests/test_peppol_messages.py

```diff
@@ -134,10 +134,11 @@ def _get_mock_data(cls, error=False, nr_invoices=1):
 
     @contextmanager
     def _set_context(self, other_context):
-        previous_context = self.env.context
-        self.env.context = dict(previous_context, **other_context)
-        yield self
-        self.env.context = previous_context
+        cls = self.__class__
+        env = cls.env
+        env = env(context=dict(env.context, **other_context))
```

Comment:
Please avoid using the same variable for two different things.  And optimizing `cls.env` looks ridiculous to me...
```suggestion
        env = cls.env(context=dict(cls.env.context, **other_context))
```